### PR TITLE
Add Resources to Cloudflare Terraform Provider

### PR DESCRIPTION
**Pull request by apiazza-dd**
This PR allows customers to provision resources for their Cloudflare Accounts using Terraform.

### CODE REVIEW

File: docs/resources/integration_cloudflare_account.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ resource "datadog_integration_cloudflare_account" "foo" {
 ### Optional
 
 - `email` (String) The email associated with the Cloudflare account. If an API key is provided (and not a token), this field is also required.
+- `resources` (List of String) An allowlist of resources to restrict pulling metrics for including 'web', 'dns', 'lb' (load balancer), 'worker')
```
Comment (maycmlee) [inline docs/resources/integration_cloudflare_account.md:36]: Are `web`, `dns`, etc. tags on the metrics? Or something else?

```suggestion
- `resources` (List of String) An allowlist of resources to restrict pulling metrics for including `web`, `dns`, `lb` (load balancer), 'worker')
```

**Review by maycmlee — COMMENTED** [review]
Small suggestions and a question.

File: docs/resources/integration_cloudflare_account.md (hunk 1 above)
Comment (apiazza-dd) [reply docs/resources/integration_cloudflare_account.md:36]: These are enabling which resources to collect when provisioning a cloudflare account via terraform.

Comment (apiazza-dd) [reply docs/resources/integration_cloudflare_account.md:36]: These docs are also autogenerated

File: docs/resources/integration_cloudflare_account.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ resource "datadog_integration_cloudflare_account" "foo" {
 ### Optional
 
 - `email` (String) The email associated with the Cloudflare account. If an API key is provided (and not a token), this field is also required.
+- `resources` (List of String) An allowlist of resources to restrict pulling metrics for including `web`, `dns`, `lb` (load balancer), `worker`)
```
Comment (maycmlee) [inline docs/resources/integration_cloudflare_account.md:36]:
```suggestion
- `resources` (List of String) An allowlist of resources, such as `web`, `dns`, `lb` (load balancer), `worker`, to pull metrics for.
```

**Review by maycmlee — COMMENTED** [review]
How about this?

**Review by alpinet — DISMISSED** [review]

File: docs/resources/integration_cloudflare_account.md (hunk 2 above)
Comment (apiazza-dd) [reply docs/resources/integration_cloudflare_account.md:36]: I think the restriction part is important here, by default, they're set to all, but if they want one turned off, then they use the resources list. Phrasing as an allow list makes it seem like if they specify one, they'll collect metrics for it, when it's the opposite.

Comment (maycmlee) [reply docs/resources/integration_cloudflare_account.md:36]: @apiazza-dd Ah okay, thanks for the clarification. I updated the suggestion, what do you think?

Comment (maycmlee) [reply docs/resources/integration_cloudflare_account.md:36]: @apiazza-dd actually rereading your comment, I might have misunderstood.

> but if they want one turned off, then they use the resources list. 

Is the allowlist of resources, the resources that users want to pull metrics for or the ones they don't want pulling metrics for?

**Review by ZacharyAnderson — DISMISSED** [review]

File: docs/resources/integration_cloudflare_account.md (hunk 2 above)
Comment (apiazza-dd) [reply docs/resources/integration_cloudflare_account.md:36]: @maycmlee These docs are autogenerated and have been updated to match our other repository. Does this look okay, or should I make a change in the other repo?

Comment (maycmlee) [reply docs/resources/integration_cloudflare_account.md:36]: DM'd. Docs are autogenerated so the suggestion will be updated in the other repo.

**Review by maycmlee — APPROVED** [review]

**Review by ZacharyAnderson — DISMISSED** [review]

File: docs/resources/integration_cloudflare_account.md (hunk 2 above)
Comment (apiazza-dd) [reply docs/resources/integration_cloudflare_account.md:36]: https://github.com/DataDog/datadog-api-spec/pull/3270/files

**Review by gunterd — APPROVED** [review]

**Review by maycmlee — DISMISSED** [review]

**Review by gunterd — APPROVED** [review]

**Review by gunterd — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**